src/main/java/net/openhft/chronicle/wire/JSONWire.java

```diff
@@ -854,10 +900,17 @@ public JSONWire typeLiteral(@Nullable CharSequence type) {
         @Override
         public JSONValueOut typePrefix(@NotNull CharSequence typeName) {
             if (useTypes) {
-                startBlock('{');
+                boolean nested = bytes.peekUnsignedByte(bytes.writePosition() - 1) == '{';
```

Comment:
Could `bytes.writePosition()-1` yield a value that breaks `peekUnsignedByte`?

Reply:
peekUnsignedByte returns -1 if out of bounds.